src/eedl/image.py

```diff
@@ -326,8 +326,8 @@ def zonal_stats(self,
 						print(i)
 
 	def _check_task_status(self) -> Dict[str, Union[Dict[str, str], bool]]:
-
-		new_status = self.task.status()
+		if self.task is not None:
```

Comment:
So, I think it's good to check this, but if self.task is None, we have bigger problems I think, and this shouldn't just silently ignore it (it'll also cause an exception later if self.task IS None, because new_status won't be defined on line 333.

Was this change to accommodate type checking, or something else? We should rethink or more completely flesh this check out.

Reply:
Yeah, mypy throws an error without that check: "error: Item "None" of "Any | None" has no attribute "status"  [union-attr]". This is a consequence of defining task before it takes a value. Either PyCharm or flake8 wants things defined at the top of the class, most things are type-hinted as Optional meaning they could be None, hence the error

Reply:
OK. self.task should always be defined by the time this function is called. If we want to:
1. Make sure to define the task as None when we create the class (which is sensible enough) and
2. satisfy mypy

Then we should raise some kind of error if task is None. The rest of that function can't proceed if task is None. And if Task is None, then it means we didn't properly start an export and we shouldn't be calling this function - bigger things are wrong if task is None and we can't recover from within this function.